binder/src/main/java/io/grpc/binder/SecurityPolicies.java

```diff
@@ -116,6 +135,39 @@ public Status checkAuthorization(int uid) {
     };
   }
 
+  /**
+   * Creates {@link SecurityPolicy} which checks if the SHA-256 hash of the package signature
+   * matches any of {@code requiredSignatureSha256Hashes}.
+   *
+   * @param packageName the package name of the allowed package.
+   * @param requiredSignatureSha256Hashes the SHA-256 digests of the signatures of the allowed
+   *     package.
+   * @throws NullPointerException if any of the inputs are {@code null}.
+   * @throws IllegalArgumentException if {@code requiredSignatureSha256Hashes} is empty, or if any
+   *     of the {@code requiredSignatureSha256Hashes} are not of length 32.
+   */
+  public static SecurityPolicy oneOfSignatureSha256Hash(
+      PackageManager packageManager,
+      String packageName,
+      ImmutableList<byte[]> requiredSignatureSha256Hashes) {
+    Preconditions.checkNotNull(packageManager);
+    Preconditions.checkNotNull(packageName);
+    Preconditions.checkNotNull(requiredSignatureSha256Hashes);
+    Preconditions.checkArgument(!requiredSignatureSha256Hashes.isEmpty());
+    for (byte[] requiredSignatureSha256Hash : requiredSignatureSha256Hashes) {
```

Comment:
We should copy the `byte[]`s, since they could be mutated. Some times we might just say "don't change them" in the documentation, but this is security related, done once at startup, and seems like it'd have relatively few entries. So it seems better to me to just copy. I don't know how @markb74 feels about it though.

Reply:
wondering even using ImmutableList.copyOf(byte[]s) still can change the byte in byte[] of the original collection during iteration? is it a deep copy?

Reply:
`byte[]` itself is mutable. ImmutableList is only one-level of immutability; it is not guaranteed to be transitive.

Reply:
The byte[]s are still not being copied.